src/components/UserActions/UserActivity.ts

```diff
@@ -103,126 +93,167 @@ export class UserActivity extends Component {
+            return { caseSubmitSessionIndex, caseSubmitSession };
+        }
+
+        // If we didn't, we can try to find a session that occured just before the ticket create.
+        const potentialSessionBeforeIndex = this.sessions.findIndex(
+            (session) => session.actions[0]?.timestamp <= this.options.ticketCreationDateTime
+        );
+
+        if (potentialSessionBeforeIndex !== -1) {
+            if (this.isPartOfTheSameSession(this.sessions[potentialSessionBeforeIndex].actions[0], this.options.ticketCreationDateTime)) {
+                return { caseSubmitSessionIndex: potentialSessionBeforeIndex, caseSubmitSession: this.sessions[potentialSessionBeforeIndex] };
+            }
+
+            // If the session before the ticket create is not part of the same session, create a standalone session.
+            this.sessions.splice(potentialSessionBeforeIndex, 0, new UserActionSession(this.options.ticketCreationDateTime, []));
+
+            return { caseSubmitSessionIndex: potentialSessionBeforeIndex, caseSubmitSession: this.sessions[potentialSessionBeforeIndex] };
+        }
+        return { caseSubmitSessionIndex: -1, caseSubmitSession: null };
+    }
 
-        return isClick || (useInclude && isIncluded) || (!useInclude && !isExcluded);
+    private findSurroundingSessions(): UserActionSession[] {
+        return this.sessions.slice(Math.max(0, this.caseSubmitSessionIndex - 2), Math.min(this.caseSubmitSessionIndex + 3, this.sessions.length));
     }
 
     private render() {
+        this.element.innerHTML = '';
+
         const panel = document.createElement('div');
         panel.classList.add(MAIN_CLASS);
 
-        const timestampSection = document.createElement('div');
-        timestampSection.classList.add(HEADER_CLASS);
-
-        this.buildTimestampSection().forEach((el) => timestampSection.appendChild(el));
-
         const activitySection = this.buildActivitySection();
         activitySection.classList.add(ACTIVITY_CLASS);
 
-        panel.appendChild(timestampSection);
         panel.appendChild(activitySection);
-
-        this.element.innerHTML = '';
         this.element.appendChild(panel);
     }
 
     private buildActivitySection(): HTMLElement {
         const list = document.createElement('ol');
 
-        this.buildListItems(this.actions).forEach((listItem, index, array) => {
-            list.appendChild(listItem);
-            if (index < array.length - 1) {
-                list.appendChild(this.buildBubble());
+        this.buildSessionsItems(this.sessionsToDisplay).forEach((sessionItem) => {
+            if (sessionItem) {
+                list.appendChild(sessionItem);
             }
         });
 
         return list;
     }
 
-    private buildBubble() {
-        const li = document.createElement('li');
-        li.classList.add(BUBBLE_CLASS);
-        return li;
-    }
+    private buildSessionsItems(sessions: UserActionSession[]): HTMLElement[] {
+        let hitExpanded = false;
```

Comment:
A session has it's own "expanded" value, basically we start with the session that contains the case creation.

Any session before is collapsed, and any session after is collapsed at the start. Because the text of the button at the behavior changes after we've hit the expanded session, I need to keep track of it.